libnavui-app/src/main/java/com/mapbox/navigation/ui/app/internal/controller/RouteAlternativeStateController.kt

```diff
@@ -0,0 +1,82 @@
+package com.mapbox.navigation.ui.app.internal.controller
+
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.base.route.NavigationRoute
+import com.mapbox.navigation.base.route.RouterOrigin
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.core.internal.extensions.flowRouteAlternativeObserver
+import com.mapbox.navigation.ui.app.internal.Action
+import com.mapbox.navigation.ui.app.internal.State
+import com.mapbox.navigation.ui.app.internal.Store
+import com.mapbox.navigation.ui.app.internal.navigation.NavigationState
+import com.mapbox.navigation.ui.app.internal.routefetch.RoutePreviewAction
+import com.mapbox.navigation.ui.app.internal.routefetch.RoutesAction
+import com.mapbox.navigation.utils.internal.ifNonNull
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.launch
+
+@OptIn(ExperimentalPreviewMapboxNavigationAPI::class)
+class RouteAlternativeStateController(private val store: Store) : StateController() {
```

Comment:
I think you can extend `UIComponent` instead of `StateController`, because as a `Reducer` this class does nothing.